generate-map.py

```diff
@@ -3,38 +3,84 @@
 import sqlite3
 import json
 from geopy.geocoders import Nominatim
+import os
 
 
 def getUserInformation():
-    geolocator = Nominatim(timeout=10, user_agent='Opencast map generator')
+
+    locations = []
 
     cur = sqlite3.connect('user.db').cursor()
     cur.execute('select distinct country, city, organization from user')
     for country, city, organization in cur.fetchall():
-        location = geolocator.geocode({'country': country, 'city': city},
-                                      addressdetails=True) \
-                   or geolocator.geocode('%s, %s' % (country, city),
-                                         addressdetails=True)
-        print((country, city), '->', location)
-        if location:
-            yield (location.latitude, location.longitude, organization)
+        if organization != "None":
+            locations.append(compareCache(country, city, organization))
+    return locations
+
+
+def getGeoCode(country, city, organization, dataList):
+
+    geolocator = Nominatim(timeout=10, user_agent='Opencast map generator')
+    newLocation = geolocator.geocode(country, city, addressdetails=True) \
+        or geolocator.geocode('%s, %s' % (country, city), addressdetails=True)
+    if newLocation:
+        os.remove("cache.json")
+        f = open("cache.json", "a")
+        geoLocation = {"country": country, "city": city,
+                       "organization": organization,
+                       "latitude": newLocation.latitude,
+                       "longitude": newLocation.longitude}
+        dataList.append(geoLocation)
+        f.write(json.dumps(dataList))
+
+        geoLocation = {"country": country,
+                       "city": city, "organization": organization,
+                       "latitude": newLocation.latitude,
+                       "longitude": newLocation.longitude}
+        return geoLocation
 
 
 def convertGeoJson(addresses):
+
     features = []
-    for lat, lon, org in addresses:
-        features.append({
-            "type": "Feature",
-            "properties": {
-                'institution': org,
-            },
-            "geometry": {
-                "type": "Point",
-                "coordinates": [lon, lat]
-            }})
+    for i in addresses:
+        if i is not None:
+            print(i)
+            features.append({
+                "type": "Feature",
+                "properties": {
+                    'institution': i["organization"]
+                },
+                "geometry": {
+                    "type": "Point",
+                    "coordinates": [i["longitude"], i["latitude"]]
+                }})
     return {"type": "FeatureCollection", "features": features}
 
 
+def compareCache(country, city, organization):
+
+    check = 0
+    with open('cache.json') as data_file:
+        if os.stat("cache.json").st_size == 0:
```

Comment:
It's easier to ask for forgiveness than for permission ([EAFP](https://docs.python.org/3.4/glossary.html?highlight=eafp#term-eafp)) is a Python principle you should use here since it would make the code much simpler.

This would result in code like this:

```python
with open('cache.json') as data_file:
    try:
        … = json.load(data_file)
        # everything worked
    except FileNotFoundError:
        return getGeoCode(country, city, organization, data)
```

Reply:
Your version looks more understandable and much simpler. Thanks for this suggestion!